opal/mca/reachable/netlink/configure.m4

```diff
@@ -71,74 +72,37 @@ AC_DEFUN([OPAL_REACHABLE_NETLINK_CHECK_LIBNL3],[
-#endif
-                        ]],[[/* empty body */]])],
-                    [HAVE_LIBNL3=1],    dnl our program compiled
-                    [HAVE_LIBNL3=0])],  dnl our program failed to compile
-                [HAVE_LIBNL3=0],  dnl AC_CHECK_HEADER failed
-                [#include <netlink/netlink.h>
-                ])],
-             [AC_MSG_RESULT([missing])
-              HAVE_LIBNL3=0])  dnl "/usr/include/libnl3" does not exist
-
-       # nl_recvmsgs_report is a symbol that is only present in v3
-       AS_IF([test "$HAVE_LIBNL3" -eq 1],
-             [AC_SEARCH_LIBS([nl_recvmsgs_report], [nl-3],
-                             [# We also need libnl-route-3
-                              AC_SEARCH_LIBS([nl_rtgen_request], [nl-route-3],
-                                             [$2="-lnl-3 -lnl-route-3"
-                                              HAVE_LIBNL3=1],
-                                             [HAVE_LIBNL3=0])],
-                             [HAVE_LIBNL3=0])])
-
-       AS_IF([test "$HAVE_LIBNL3" -eq 1],
-             [AC_MSG_NOTICE([using libnl-3])],
-             [# restore $1 since we are falling back to libnl (v1)
-              $1=""
-              AC_SEARCH_LIBS([nl_connect], [nl],
-                             [$2="-lnl"],
-                             [AC_MSG_WARN([Cannot find libnl-3 nor libnl])
-                              AS_IF([test "$3" = "1"],
-                                    [AC_MSG_ERROR([Cannot continue])])
-                             ])
-              AC_MSG_NOTICE([using libnl (v1)])])
-
-       # libnl_utils.h does not include configure-generated config.h,
-       # so it may not see the HAVE_LIBNL3 #define.  Hence, we set
-       # HAVE_LIBNL3 as both a C preprocessor macro (in case some
-       # other file includes config.h before libnl_utils.h) and a
-       # Makefile macro (so that the app can set HAVE_LIBNL3 via
-       # CPPFLAGS).  Also, this macro may be used in multiple
-       # different libraries; setting HAVE_LIBNL3 both ways lets the
-       # application choose which way to set it.
-       AC_SUBST([HAVE_LIBNL3])
-       AC_DEFINE_UNQUOTED([HAVE_LIBNL3],[$HAVE_LIBNL3],
-                          [set to 1 if should use libnl v3, set to 0 for libnl v11])
-
-       LIBS=$save_LIBS
-       AS_UNSET([save_LIBS])
-       CPPFLAGS=$save_CPPFLAGS
-       AS_UNSET([save_CPPFLAGS])
+	# Default to a numeric value (this value gets AC_DEFINEd)
+	HAVE_LIBNL3=0
+
+	###################################################
+	# NOTE: We *must* check for libnl3 before libnl.
+	###################################################
+
+	AS_IF([test "$opal_want_libnl" != "no" &&
+               test $opal_libnl_version -ne 1],
+	      [OPAL_CHECK_LIBNL3([$opal_libnl_location], [opal_nl])])
+	AS_IF([test "$opal_want_libnl" != "no" &&
+               test $opal_libnl_version -ne 3 &&
+	       test "$opal_nl_LIBS" = ""],
+	      [OPAL_CHECK_LIBNL([$opal_libnl_location], [opal_nl])])
+
```

Comment:
I'm not sure I understand the logic of the 2 preceding AS_IFs.

- We probably shouldn't even reach here if `$opal_want_libnl = no`.
- You check `$opal_libnl_version -ne 1` in the first one, and `-ne 3` in the second one.  Why is this?  Is this to check to see if some other part of the code base has already determined if we are using libnl v1 or v3?  If so, won't `$opal_nl` (or, `$opal_reachable_netlink` when it's renamed :wink:) be empty, and then the AS_IF in the next line will fail?